esphome/components/abbaurora/abbaurora.h

```diff
@@ -0,0 +1,259 @@
+  bool read_last_four_alarms_();
+  bool read_manufacturing_week_year_();
+  bool read_state_();
+  bool read_system_info_central_(uint8_t var);
+  bool read_system_partnumber_();
+  bool read_system_partnumber_central_();
+  bool read_system_serialnumber_();
+  bool read_system_serialnumber_central_();
+  bool read_timedate_();
+  bool read_version_();
+  bool send_(uint8_t address, uint8_t param0, uint8_t param1, uint8_t param2, uint8_t param3, uint8_t param4,
+             uint8_t param5, uint8_t param6);
+  bool write_baudrate_setting_(uint8_t baudcode);
+
+  static std::string transmission_state_text(uint8_t id);
+  static std::string global_state_text(uint8_t id);
+  static std::string dcdc_state_text(uint8_t id);
+  static std::string inverter_state_text(uint8_t id);
+  static std::string alarm_state_text(uint8_t id);
+
+  union {
+    uint8_t asBytes[4];
+    float asFloat;
+  } float_bytes_;
+
+  union {
+    uint8_t asBytes[4];
+    uint32_t asUlong;
+  } long_bytes_;
+
+  using DataState = struct {
+    uint8_t TransmissionState;
+    uint8_t GlobalState;
+    uint8_t InverterState;
+    uint8_t Channel1State;
+    uint8_t Channel2State;
+    uint8_t AlarmState;
+    bool ReadState;
+  };
+  DataState state_;
+
+  using DataVersion = struct {
+    uint8_t TransmissionState;
+    uint8_t GlobalState;
+    std::string Par1;
+    std::string Par2;
+    std::string Par3;
+    std::string Par4;
+    bool ReadState;
+  };
+  DataVersion data_version_;
+
+  using DataDSP = struct {
+    uint8_t TransmissionState;
+    uint8_t GlobalState;
+    float Value;
+    bool ReadState;
+  };
+  DataDSP dsp_;
+
+  using DataTimeDate = struct {
+    uint8_t TransmissionState;
+    uint8_t GlobalState;
+    uint32_t Seconds;
+    bool ReadState;
+  };
+  DataTimeDate timedate_;
+
+  using DataLastFourAlarms = struct {
+    uint8_t TransmissionState;
+    uint8_t GlobalState;
+    uint8_t Alarms1;
+    uint8_t Alarms2;
+    uint8_t Alarms3;
+    uint8_t Alarms4;
+    bool ReadState;
+  };
+  DataLastFourAlarms last_four_alarms_;
+
+  // Inverters
+  using DataSystemPartNumber = struct {
+    std::string PartNumber;
+    bool ReadState;
+  };
+  DataSystemPartNumber system_partnumber_;
+
+  using DataSystemSerialNumber = struct {
+    std::string SerialNumber;
+    bool ReadState;
+  };
+  DataSystemSerialNumber system_serialnumber_;
+
+  using DataManufacturingWeekYear = struct {
+    uint8_t TransmissionState;
+    uint8_t GlobalState;
+    std::string Week;
+    std::string Year;
+    bool ReadState;
+  };
+  DataManufacturingWeekYear manufacturing_week_year_;
+
+  using DataFirmwareRelease = struct {
+    uint8_t TransmissionState;
+    uint8_t GlobalState;
+    std::string Release;
+    bool ReadState;
+  };
+  DataFirmwareRelease firmware_release_;
+
+  using DataCumulatedEnergy = struct {
+    uint8_t TransmissionState;
+    uint8_t GlobalState;
+    uint32_t Energy;
+    bool ReadState;
+  };
+  DataCumulatedEnergy cumulated_energy_;
```

Comment:
```suggestion
  struct DataCumulatedEnergy {
    uint8_t transmission_state;
    uint8_t global_state;
    uint32_t energy;
    bool read_state;
  } cumulated_energy_;
```

Not sure why clang-tidy isn't complaining about this, but our naming conventions require `lower_snake_case` (same for all other structs here)